Marlin/Conditionals_post.h

```diff
@@ -406,9 +406,13 @@
     #endif
   #endif
 
-  #define IS_Z2_OR_PROBE(P) (   (PIN_EXISTS(Z2_MIN)      && (P) == Z2_MIN_PIN)       \
-                             || (PIN_EXISTS(Z2_MAX)      && (P) == Z2_MAX_PIN)       \
-                             || (PIN_EXISTS(Z_MIN_PROBE) && (P) == Z_MIN_PROBE_PIN))
+  /**
+   *  Deciding which endstops are active at pre-processor time is "interesting"
+   *      all real pins are fully useable, when Z2_MIN_PIN or Z2_MAX_PIN exist they don't have a value (all compares fail)
+   */
```

Comment:
What is meant by this? :

"When Z2_MIN_PIN or Z2_MAX_PIN exist they don't have a value (all compares fail)"

Does that mean that the following would produce an error?:
```cpp
#if !defined(Z2_MIN_PIN) || !Z2_MIN_PIN
  #error "Z2_MIN_PIN should be assigned but is zero!"
#endif
```